pkg/kubecost/etlrange.go

```diff
@@ -0,0 +1,79 @@
+package kubecost
+
+import (
+	"fmt"
+	"github.com/kubecost/opencost/pkg/util/json"
+	"sync"
+)
+
+// SetRange is a generic implementation of the SetRanges that act as containers. It covers the basic functionality that
+// is shared by the basic types but is meant to be extended by each implementation.
+type SetRange[T ETLSet] struct {
+	lock sync.RWMutex
+	sets []T
+}
+
+// Append attaches the given ETLSet to the end of the sets slice.
+// currently does not check that the window is correct.
+func (r *SetRange[T]) Append(that T) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	r.sets = append(r.sets, that)
+}
+
+// Each invokes the given function for each ETLSet in the SetRange
+func (r *SetRange[T]) Each(f func(int, T)) {
+	if r == nil {
+		return
+	}
+
+	for i, set := range r.sets {
+		f(i, set)
+	}
+}
```

Comment:
@mbolt35 discussed pros and cons of Set(Range) locking a little while ago, but unfortunately I don't remember where our conversations landed -- should `Each` lock?